### PR TITLE
feat(cq): add support for cq and comp_channel

### PR DESCRIPTION
**Pull request by FujiZ**
1. implement generic builder for both cq and cq_ex
2. implement comp_channel and associate its lifetime with cq

### CODE REVIEW

**Comment by FujiZ** [comment]
Maybe we can put `CompletionQueue` and `CompletionChannel` in a single module called `completion`.

**Comment by dragonJACson** [comment]
> Maybe we can put `CompletionQueue` and `CompletionChannel` together in a single module called `completion`.

I agree, as `CompletionChannel` would only be a **part** of `CompletionQueue`? So both of them and some other attributes related with CQ should be in a single module

File: examples/test_cq.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+use sideway::verbs::completion_queue;
+use sideway::verbs::device;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let device_list = device::DeviceList::new()?;
+    for device in &device_list {
+        let ctx = device.open().unwrap();
+
+        let comp_channel = comp_channel::CompletionChannel::new(&ctx).unwrap();
+        let mut builder = completion_queue::CompletionQueueBuilder::new(&ctx);
```
Comment (dragonJACson) [inline examples/test_cq.rs:11]: What about using `ctx.create_comp_channel` `ctx.create_cq_builder`? So that we don't need to pass the argument explicitly.

Comment (FujiZ) [reply examples/test_cq.rs:11]: Yeah we can add some methods at context level to invoke these implementations.

**Review by dragonJACson — APPROVED** [review]

**Comment by FujiZ** [comment]
Ref #8

**Pull request merged**